token/transfer-hook/interface/src/offchain.rs

```diff
@@ -68,3 +76,236 @@ where
 
     Ok(())
 }
+
+/// Offchain helper to get all additional required account metas for an execute
+/// instruction, based on a validation state account.
+///
+/// To be client-agnostic and to avoid pulling in the full solana-sdk, this
+/// simply takes a function that will return its data as `Future<Vec<u8>>` for
+/// the given address. Can be called in the following way:
+///
+/// ```rust,ignore
+/// add_extra_account_metas_for_execute(
+///     &mut instruction,
+///     &program_id,
+///     &source,
+///     &mint,
+///     &destination,
+///     &authority,
+///     amount,
+///     |address| self.client.get_account(&address).map_ok(|opt| opt.map(|acc| acc.data)),
+/// )
+/// .await?;
+/// ```
+#[allow(clippy::too_many_arguments)]
+pub async fn add_extra_account_metas_for_execute<F, Fut>(
+    instruction: &mut Instruction,
+    program_id: &Pubkey,
```

Comment:
I think I'd rather add a comment for this, since technically this function can stand on its own aside from creating `ExecuteInstruction` from a transfer hook context. We also call it `program_id` in `instruction::execute(..)`. Wdyt?

Reply:
a comment works too!